Tests/SonarQube.TeamBuild.PostProcessor.Tests/Infrastructure/MockCodeCoverageProcessor.cs

```diff
@@ -7,42 +7,64 @@
 using SonarQube.TeamBuild.Integration;
 using SonarQube.Common;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
 
 namespace SonarQube.TeamBuild.PostProcessor.Tests
 {
     internal class MockCodeCoverageProcessor : ICoverageReportProcessor
     {
-        private bool methodCalled;
+        private bool processCoverageMethodCalled;
+        private bool initalizedCalled;
 
         #region Test helpers
 
-        public bool ValueToReturn { get; set; }
+        public bool ProcessValueToReturn { get; set; }
+        public bool InitialiseValueToReturn { get; set; }
 
         #endregion
 
         #region ICoverageReportProcessor interface
 
-        public bool ProcessCoverageReports(AnalysisConfig context, TeamBuildSettings settings, ILogger logger)
+        public bool Initialise(AnalysisConfig context, TeamBuildSettings settings, ILogger logger)
         {
-            Assert.IsFalse(this.methodCalled, "Expecting ProcessCoverageReports to be called only once");
-            this.methodCalled = true;
-            return ValueToReturn;
+            Assert.IsFalse(this.initalizedCalled, "Expecting Initialise to be called only once");
+            this.initalizedCalled = true;
+            return InitialiseValueToReturn;
+        }
+
+        public bool ProcessCoverageReports()
+        {
+            Assert.IsFalse(this.processCoverageMethodCalled, "Expecting ProcessCoverageReports to be called only once");
+            Assert.IsTrue(this.initalizedCalled, "Expecting Initialise to be called first");
+            this.processCoverageMethodCalled = true;
+            return ProcessValueToReturn;
         }
 
         #endregion
 
         #region Checks
 
-        public void AssertExecuted()
+        public void AssertExecutedCalled()
+        {
+            Assert.IsTrue(this.processCoverageMethodCalled, "Expecting the sonar-runner to have been called");
+        }
+
+        public void AssertExecutedNotCalled()
```

Comment:
Thank you. I updated the PR.

Bogdan

From: duncanpMS [mailto:notifications@github.com]
Sent: 01 September 2015 10:56
To: SonarSource/sonar-msbuild-runner <sonar-msbuild-runner@noreply.github.com>
Cc: Bogdan Gavril <bgavril@microsoft.com>
Subject: Re: [sonar-msbuild-runner] SONARMSBRU-130 - Do not crash and do not stop the analysis if VS is not installed on the machine (#108)


In Tests/SonarQube.TeamBuild.PostProcessor.Tests/Infrastructure/MockCodeCoverageProcessor.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fSonarSource%2fsonar-msbuild-runner%2fpull%2f108%23discussion_r38402813&data=01%7c01%7cBGAVRIL%40064d.mgd.microsoft.com%7cabc4480170a34e42713308d2b2b395f6%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=4nUxHZ8m11WUBvNKk5QtiY2qh%2fCLHwtx3nLL%2fDLjvN4%3d>:

>          }

>

>          #endregion

>

>          #region Checks

>

> -        public void AssertExecuted()

> +        public void AssertExecutedCalled()

> +        {

> +            Assert.IsTrue(this.processCoverageMethodCalled, "Expecting the sonar-runner to have been called");

> +        }

> +

> +        public void AssertExecutedNotCalled()

FYI Code coverage requires VS2013 Ultimate or Premium / VS2015 Enterprise.
It isn't available in VS2013 Professional.
See https://msdn.microsoft.com/en-us/library/dd537628(v=vs.120).aspx<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fmsdn.microsoft.com%2fen-us%2flibrary%2fdd537628(v%3dvs.120).aspx&data=01%7c01%7cBGAVRIL%40064d.mgd.microsoft.com%7cabc4480170a34e42713308d2b2b395f6%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=71CYyx6ETJBJLrmW46jVJebQ2bETZcxBxrSnNCGGm%2fE%3d>

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fSonarSource%2fsonar-msbuild-runner%2fpull%2f108%2ffiles%23r38402813&data=01%7c01%7cBGAVRIL%40064d.mgd.microsoft.com%7cabc4480170a34e42713308d2b2b395f6%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=npUhTMECYxv2nnAbYwpjZFxL6pzWjIlf7fYYmyx%2bKL8%3d>.